test/request.js

```diff
@@ -131,17 +129,19 @@ describe('Request', () => {
 
         const handler = (request) => {
 
-            let expectedClientAddress = '127.0.0.1';
-            if (Net.isIPv6(server.listener.address().address)) {
-                expectedClientAddress = '::ffff:127.0.0.1';
+            // Call twice to reuse cached values
+
+            if (Common.hasIPv6) {
+                // ::ffff:127.0.0.1 on node v14 and v16, ::1 on node v18.
+                expect(request.info.remoteAddress).to.match(/^::ffff:127\.0\.0\.1|::1$/);
+                expect(request.info.remoteAddress).to.match(/^::ffff:127\.0\.0\.1|::1$/);
```

Comment:
Would it make sense to convert these to regular IPv4 adresses in hapi before assigning to `remoteAddress`? The property is not intended to signal the IP layer protocol, so I think it would be fine.

The mechanics of the address are from: https://datatracker.ietf.org/doc/html/rfc3493#section-3.7

Reply:
That seems reasonable.  It may be a little hard to test if there's a code branch relying on IPv6 networking to hit.  That aside for a moment, do you find this to be rigorous enough?
```js
const socketAddress = this._request.raw.req.socket.remoteAddress;
this._remoteAddress = socketAddress.startsWith('::ffff:') ?
    socketAddress.slice('::ffff:'.length) :
    socketAddress;
```

If after some back and forth this proves complex, I might lean towards continuing to pass through `socket.remoteAddress`.

Reply:
Thanks for the feedback, this is ready for another look.